File: LeetCode/3010/3010.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
        pair<int,int> temp{60,60};
        for(auto iter = nums.begin()+1; iter!= nums.end(); ++iter)
        {
            if((*iter) < temp.first) temp.second = std::exchange(temp.first, (*iter));
            else if(*iter < temp.second) temp.second = (*iter);
        }
        return *nums.begin() + temp.first + temp.second;
    }
};

File: LeetCode/3010/3010.rs
impl Solution {
    pub fn minimum_cost(nums: Vec<i32>) -> i32 {
        let mut answer = nums[0];
        let mut pair = (60,60);
        nums[1..].iter().for_each(|&x| {
            if x <= pair.0 {pair.1 = pair.0; pair.0 = x;}
            else if x <= pair.1 {pair.1 = x;}
        });
        answer += pair.1 + pair.0;
        answer
    }
}
